src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -73,26 +78,99 @@ private boolean isFileNameMatching(TreeWalkerAuditEvent event) {
     private boolean isLineMatching(TreeWalkerAuditEvent event) {
         boolean result = false;
         if (event.getLocalizedMessage() != null) {
-            for (List<Integer> singleLineRangeList : lineRangeList) {
-                final int startLine = singleLineRangeList.get(0) + 1;
-                final int endLine = singleLineRangeList.get(1) + 1;
-                final int currentLine = event.getLine();
-                result = currentLine >= startLine && currentLine < endLine;
-                if (result) {
-                    break;
-                }
+            result = lineMatching(event.getLine());
+        }
+        return result;
+    }
+
+    private boolean lineMatching(int currentLine) {
+        boolean result = false;
+        for (List<Integer> singleLineRangeList : lineRangeList) {
+            final int startLine = singleLineRangeList.get(0) + 1;
+            final int endLine = singleLineRangeList.get(1) + 1;
+            result = currentLine >= startLine && currentLine < endLine;
+            if (result) {
+                break;
             }
         }
         return result;
     }
 
     /**
-     * Is matching by xpath query.
+     * Is matching by context strategy.
      *
      * @param event event
      * @return true if it is matching or not set.
      */
-    private boolean isXpathQueryMatching(TreeWalkerAuditEvent event) {
-        return true;
+    private boolean isContextStrategyCheck(TreeWalkerAuditEvent event) {
+        boolean result = false;
+        final String checkShortName = getCheckShortName(event);
+        if ("OuterTypeNumberCheck".equals(checkShortName)) {
```

Comment:
OuterTypeNumberCheck logs violations on root AST, so you can still check its child nodes to check whether it matches or not. I just don't like that we have different logic for different Checks.

Reply:
yes I know, at the beginning I also put `OuterTypeNumberCheck` to check its child nodes to check whether it matches or not, but I found that `OuterTypeNumberCheck`'s root AST has a problem. for example:

```
PACKAGE_DEF -> package [1:0]
|--ANNOTATIONS -> ANNOTATIONS [1:18]
|--DOT -> . [1:18]
|   |--IDENT -> TreeWalker [1:8]
|   `--IDENT -> OuterTypeNumber [1:19]
`--SEMI -> ; [1:34]
CLASS_DEF -> CLASS_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|   |--SINGLE_LINE_COMMENT -> // [1:37]
|   |   `--COMMENT_CONTENT ->  violation context\n [1:39]
|   `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_CLASS -> class [3:7]
|--IDENT -> Test [3:13]
`--OBJBLOCK -> OBJBLOCK [3:18]
    |--LCURLY -> { [3:18]
    `--RCURLY -> } [4:0]
```

Reply:
done